### PR TITLE
Add `ObjectPool<StringBuilder>` to default services

### PR DESCRIPTION
**Pull request by dougbu**
- can now use pooled `StringBuilder`s to address aspnet/HttpAbstractions#561 and similar issues

nit: Fix incomplete test noted when adding new one

### CODE REVIEW

File: test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs [hunk 1]
```diff
@@ -92,8 +93,27 @@ public void Build_honors_UseStartup_with_string()
...
+                var services = host.Services.GetServices<IApplicationLifetime>();
+                Assert.NotNull(services);
+                Assert.NotEmpty(services);
+
+                await AssertResponseContains(server.RequestDelegate, "Exception from constructor");
+            }
+        }
+
+        [Fact]
+        public async Task ObjectPoolOfStringBuilderRegistered_EvenWhenStartupCtorThrows()
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs]: We don't actually care about ObjectPool in the StartupThrows scenario. You can find a simpler test to copy.

**Comment by Tratcher** [comment]
Do not merge this until we see how aspnet/HttpAbstractions#561 looks.

File: test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs (hunk 1 above)
Comment (dougbu) [inline test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs]: Suggestions? I don't see much that just validates services are registered.

Comment (Tratcher) [inline test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs]: `.Configure(app => { })` would be an adequate replacement for UseStartup in this test.

**Comment by dougbu** [comment]
:up::date: and aspnet/HttpAbstractions#587 is out.

**Comment by Tratcher** [comment]
@davidfowl Gag check?

**Comment by davidfowl** [comment]
This is odd. I'm not sure this makes sense.

**Comment by rynowak** [comment]
@davidfowl - what do you propose? Use a static pool? That might make the most sense if we're talking about HttpAbstractions/WebUtilities

**Comment by davidfowl** [comment]
Yes, I was going to suggest a static pool for something like this with optional overloads to pass in a pool into whatever method is there (low level method).

**Comment by dougbu** [comment]
>  I was going to suggest a static pool

Let's discuss this offline. We already have at least one other pattern for `ObjectPool<T>` use in other repos. Would rather move ahead w/ the general guidance on when to use what (and what to use when) in place.

**Comment by dougbu** [comment]
Will move instantiation of the pool to Kestrel as decided offline.

**Pull request closed without merging**